src/merkle/smt/mod.rs

```diff
@@ -253,3 +253,9 @@ impl From<RpoDigest> for LeafIndex<SMT_MAX_DEPTH> {
         Word::from(value).into()
     }
 }
+
+impl From<&RpoDigest> for LeafIndex<SMT_MAX_DEPTH> {
+    fn from(value: &RpoDigest) -> Self {
+        Word::from(value).into()
+    }
+}
```

Comment:
I wonder if this and the other conversions from `Word` and `RpoDigest` should actually go into `src/merkle/smt/full/mod.rs` as they seem to be relevant only for the `Smt` struct?

Reply:
They would be useful for any other sparse Merkle tree that uses 64-bit keys. As of now, we only have `Smt`, and we'll probably have that for the foreseeable future... So yes I will move them there.

Note: we might revert this depending on how we end up addressing https://github.com/0xPolygonMiden/crypto/issues/252

Reply:
fixed